clientv3/lease.go

```diff
@@ -394,6 +411,13 @@ func (l *lessor) getKeepAliveStream() pb.Lease_LeaseKeepAliveClient {
 
 func (l *lessor) switchRemoteAndStream(prevErr error) error {
 	for {
+		l.rc.mu.Lock()
+		closed := l.rc.client.cancel == nil
```

Comment:
this probably needs to use `rc.acquire()`, otherwise a `client.Close()` can happen immediately after the `Unlock()` and the `closed` value will be out of date